migration/tests.py

```diff
@@ -159,6 +162,70 @@ def test_manager_get_courses_with_limit(self):
         self.assertTrue(len(courses) <= 50)
 
 
+class WarehouseAccountManagerTestCase(unittest.TestCase):
+    """
+    Integration tests for WarehouseAccountManager class
+    """
+
+    def setUp(self) -> None:
+        wh_db_params: DBParams = {
+            "host": os.getenv('WH_HOST', ''),
+            "port": os.getenv('WH_PORT', ''),
+            "name": os.getenv('WH_NAME', ''),
+            "user": os.getenv('WH_USER', ''),
+            "password": os.getenv('WH_PASSWORD', '')
+        }
+
+        api_url: str = os.getenv('API_URL', '')
+        api_key: str = os.getenv('API_KEY', '')
+        self.api = API(api_url, api_key)
+
+        self.enrollment_term_ids: list[int] = convert_csv_to_int_list(os.getenv('ENROLLMENT_TERM_IDS', '0'))
+        self.db = DB(Dialect.POSTGRES, wh_db_params)
+        self.test_account_id = int(os.getenv('TEST_ACCOUNT_ID', 0))
+
+    def test_get_subaccount_ids(self):
+        with self.api.client:
+            manager = WarehouseAccountManager(account_id=self.test_account_id, db=self.db, api=self.api)
+            subaccount_ids = manager.get_subaccount_ids()
+        self.assertTrue(len(subaccount_ids) > 0)
+        for subaccount_id in subaccount_ids:
+            self.assertIsInstance(subaccount_id, int)
+
+    def test_manager_get_courses_in_single_term(self):
+        with self.db, self.api.client:
+            manager = WarehouseAccountManager(account_id=self.test_account_id, db=self.db, api=self.api)
+            courses = manager.get_courses_in_terms([self.enrollment_term_ids[0]], 150)
+        self.assertTrue(len(courses) > 0)
+        term_ids: list[int] = []
+        for course in courses:
+            self.assertTrue(isinstance(course, Course))
+            term_ids.append(course.enrollment_term_id)
+        term_id_set = set(term_ids)
+        self.assertTrue(len(term_id_set) == 1)
+
+    def test_manager_get_courses_in_multiple_terms(self):
+        with self.db, self.api.client:
+            manager = WarehouseAccountManager(account_id=self.test_account_id, db=self.db, api=self.api)
+            courses = manager.get_courses_in_terms(self.enrollment_term_ids)
+        self.assertTrue(len(courses) > 0)
+        term_ids: list[int] = []
+        for course in courses:
+            self.assertTrue(isinstance(course, Course))
+            term_ids.append(course.enrollment_term_id)
+        term_id_set = set(term_ids)
+        self.assertTrue(len(term_id_set) > 1)
+
+    def test_manager_get_courses_with_limit(self):
+        with self.db, self.api.client:
+            manager = WarehouseAccountManager(self.test_account_id, self.db, api=self.api)
+            courses = manager.get_courses_in_terms(self.enrollment_term_ids, 50)
```

Comment:
This test case is for getting courses with limit. 

With the data warehouse approach, is it still a need for providing the course number limit?

Reply:
Seems like it could be useful for testing if you just wanted to try fetching and/or updating a small number of courses. But also this class needs to fulfill the same interface as `AccountManager` for them to be interchangeable.